### PR TITLE
225 validate 530

### LINKED ISSUE #225: Error code 530 - A placement provider code of PR4 cannot be associated with placement P1.
#### Error code
530
#### Description to display
A placement provider code of PR4 cannot be associated with placement P1.
#### DfE Long-form description of error
Placement provider code of PR4 cannot be associated with placement P1 as these placements cannot be provided by private organisations.
#### DfE suggested coding
If `<PL>` = 'P1' then `<PLACE_PROVIDER>` must not be 'PR4'

For further info, refer to the DfE validation document [here](https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/974999/Children_looked_after_data_collection_2020_to_2021_validation_checks_-_version_1.3.pdf)

### PR DESCRIPTION
**Pull request by JRycr**
Added the 530 validation rule
Closes #225

### CODE REVIEW

File: validator903/validators.py [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
+  def _validate(dfs):
+    if 'Episodes' not in dfs:
+      return {}
+    else:
+      code_list_place =['P1']
+      code_list_place_provider=['PR4']
+
+      episodes = dfs['Episodes']
+
+      mask = episodes['PLACE'].isin(code_list_place) & episodes['PLACE_PROVIDER'].isin(code_list_place_provider)
```
Comment (mwaddoups) [inline validator903/validators.py]: You can actually simplify this slightly if you are only checking a single value - you don't need to use use `isin`. You can instead write either

```
mask = episodes['PLACE'].eq('P1') & episodes['PLACE_PROVIDER'].eq('PR4')
```

or

```
mask = (episodes['PLACE'] == 'P1') & (episodes['PLACE_PROVIDER'] == 'PR4')
```

and remove lines 14 and 15. I prefer the first approach as it has less brackets.

File: validator903/validators.py [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+        
+      validation_error_mask = mask
+      validation_error_locations = episodes.index[validation_error_mask]
+
+      return {'Episodes': validation_error_locations.tolist()}
+    
+  return error, _validate
```
Comment (mwaddoups) [inline validator903/validators.py]: A small style comment that doesn't matter for the code functionality, but is helpful for readability and consistency with the rest of the codebase:

The indentation you've used is a bit inconsistent with the rest of the code - in general we are using 4 spaces between each code block, but I think you've been using 2 spaces. Repl.it unfortunately defaults to 2 spaces so you have to remember to manually set it to 4. You can correct this for multiple lines by selecting multiple lines at once and using Tab  (or Shift+Tab to go back).

**Review by mwaddoups — DISMISSED** [review]
Nice to see this coming in Jeanette! 

A couple minor changes suggested here but I think all your code works - just highlighting some easier ways to do things and a couple style changes.

**Comment by JRycr** [comment]
Hi Mark
I have made the changes – thanks the “EQ” was what I wanted to do.

Not sure what happens when you made an amendment as have pushed and committed but then in GitHub can’t find a nice easy button saying pulling request – was easy last time. What do you do when you make an amendment?

Thanks
JEanette


Jeanette Rycroft
Senior Analyst, Joint Intelligence Unit
Customer Transformation
Wigan Council
Wigan Town Hall, Library Street, Wigan WN1 1YN

07766234730
Please note we are currently prioritising COVID 19 activity
Upcoming Leave:
2nd & 3rd September
29th September – 13th October

www.wigan.gov.uk
www.twitter.com/wigancouncil
www.facebook.com/WiganCouncilOnline
[http://www.wigan.gov.uk/pub/images/Deal2030.jpg]

**Review by dezog — APPROVED** [review]
This is looking great, sorry for the delay!

**Pull request merged**